orchagent/portsorch.cpp

```diff
@@ -2699,6 +2740,22 @@ void PortsOrch::doPortTask(Consumer &consumer)
                     }
                 }
 
+                if (tpid != 0 && tpid != p.m_tpid)
```

Comment:
How do we reset? I think the check should be just `tpid != p.m_tpid`, correct?

Reply:
In this case "tpid !=0" check serves the purpose of identifying whether it is handling non-TPID operations in which case variable tpid is initialized to "0" and there is no need to perform TPID operation.  This is inline with how MTU and speed setting is being handled.  Hope this explains why this check is required.